### ISSUE TITLE
Hotplug support

### ISSUE THREAD

**Issue by mcuee** [mcuee/libusb-win32#44]
It may be good to provide hotplug support.

As of now, libusb-win32 does not support hotplug. libusb-1.0 does not support hotplug under Windows. 
libusbK has hotplug support.

**Comment by dontech** [comment]
Please clarify what exactly you mean by hotplug.

Do you mean generating a callback when a new device is enumerated ?

**Comment by mcuee** [comment]
> Please clarify what exactly you mean by hotplug.
> 
> Do you mean generating a callback when a new device is enumerated ?

Yes, something along that line -- notification about device hotplug event.

libusbK hotplug APIs:
https://libusbk.sourceforge.net/UsbK3/group__hotk.html

libusb-1.0 hotplug APIs: (not yet implemented on Windows platform)
https://libusb.sourceforge.io/api-1.0/group__libusb__hotplug.html

**Comment by sonatique** [comment]
@mcuee : Quick question: what about fully implementing hotplug directly inside libusb-1.0 (supposing it is possible using win32-like API)?
Would there be any downside as compare to implementing the core functionality inside libusb0.sys and only calling it from libusb-1.0?

**Comment by mcuee** [comment]
I think typical hotplug implementation is in the user mode side and not on the kernel side. For example you can look at libusbK.

BTW, libusb-win32 API (libusb0.dll) is a superset of libusb-0.1, so it is not compatible with libusb-1.0 API.

**Comment by mcuee** [comment]
@dontech 

Please check if this is worth the efforts or not, if not, we can close this issue as not planned.

**Comment by dontech** [comment]
We can still provide hotplug support to libusb-1.0 in the future I think, so lets keep this one open for now.

**Comment by mcuee** [comment]
* https://github.com/libusb/libusb/pull/1406 has been merged, even though it is not the default for 1.0.30 release.
